server/plugin/client.go

```diff
@@ -531,7 +532,8 @@ func (c *client) makeHTTPRequestWithAccessToken(basePath, path, method, accessTo
 		return nil, http.StatusInternalServerError, err
 	}
 
-	req, err := http.NewRequest(method, URL, nil)
+	sanitizedURL := sanitize.URL(URL)
+	req, err := http.NewRequest(method, sanitizedURL, nil)
```

Comment:
As noted in the ticket:

> This also applies to all other API client methods and should be fixed consistently everywhere

This fix needs to be applied for all outgoing requests, but this function `makeHTTPRequestWithAccessToken` is only called by `GetUserProfile`.

Reply:
This isn't a fix at all. `mrz1836/go-sanitize` might be the first hit when you google for "golang sanitize url" but it has nothing to do with what we're trying to do here. I would not be comfortable shipping that library even if it was attempting to do the correct thing, which it's not.

Also from the ticket:

> The `organization`, `*ID` and `projectName` variables are fully user controlled and must be sanitized in case of `../` injections.

It's the input variables that need to be sanitized before they get used as URL path components, not the completed URL. The thing to look for is also named right there. This should not require any external dependencies. There are multiple possible approaches, but the stdlib `strings` and `path` packages might be helpful.

Reply:
+1 on no need for an external dependency. The fix should be applied possibly only to the `path` part of the input and also be applied to all API client methods.

Reply:
We removed the external package and used `path.Clean` to sanitize URL paths in all the needed client methods, please let us know if this looks good now